src/identity/actions/thunks/index.ts

```diff
@@ -0,0 +1,107 @@
+import {GetState} from 'src/types'
+import {Dispatch} from 'react'
+
+// Actions
+import {
+  setQuartzMe,
+  setQuartzMeStatus,
+  Actions as MeActions,
+} from 'src/me/actions/creators'
+import {
+  setCurrentOrgDetails,
+  setCurrentBillingProvider,
+  setQuartzIdentity,
+  setQuartzIdentityStatus,
+  Actions as IdentityActions,
+} from 'src/identity/actions/creators'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+// Utilities
+import {
+  fetchQuartzIdentity,
+  fetchAccountDetails,
+  fetchOrgDetails,
+} from 'src/identity/apis/auth'
+import {convertIdentityToMe} from 'src/identity/utils/convertIdentityToMe'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+// Thunks
+import {getQuartzMeThunk} from 'src/me/actions/thunks'
+
+export const getQuartzIdentityThunk = () => async dispatch => {
+  if (!isFlagEnabled('quartzIdentity')) {
+    dispatch(getQuartzMeThunk())
+    return
+  }
+
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const quartzIdentity = await fetchQuartzIdentity()
+
+    dispatch(setQuartzIdentity(quartzIdentity))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
+
+    const legacyMe = convertIdentityToMe(quartzIdentity)
+    dispatch(setQuartzMe(legacyMe, RemoteDataState.Done))
+    dispatch(setQuartzMeStatus(RemoteDataState.Done))
+  } catch (error) {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Error))
+    dispatch(setQuartzMeStatus(RemoteDataState.Error))
+    console.error(error)
+  }
+}
+
+export const getBillingProviderThunk = () => async (
+  dispatch: Dispatch<IdentityActions | MeActions>,
+  getState: GetState
+) => {
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const currentState = getState()
+    const accountId = currentState?.identity?.account?.id
+
+    const accountDetails = await fetchAccountDetails(accountId)
+
+    dispatch(setCurrentBillingProvider(accountDetails.billingProvider))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
```

Comment:
The use of the optional chaining operator is really due to paranoia here, and may be more defensive than necessary. `currentState.identity.account.id` will exist, and should always be a number, since the default value in `initialState` for `accountId` is `0`, and the only values assigned to `accountId` are primitives of type `number` returned by the `identity` endpoint.

If we wanted to be extra safe, we could throw an error to be caught by the `catch` block:

  `if (typeof accountId !== 'number') {
      throw new Error(
        'Unable to retrieve billing provider. Account ID value is not a number.'
      )
    }`